clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -478,8 +478,13 @@ boolean joinGroupIfNeeded(final Timer timer) {
 
                 resetJoinGroupFuture();
                 synchronized (AbstractCoordinator.this) {
-                    rejoinReason = String.format("rebalance failed due to '%s' (%s)", exception.getMessage(), exception.getClass().getSimpleName());
-                    rejoinNeeded = true;
+                    final String shortReason = String.format("rebalance failed due to an exception: (%s)",
+                        exception.getMessage(),
```

Comment:
@jnh5y If I understand correctly, you are saying that `exception.getMessage()` contained the stack trace. Am I right? I wonder why. Usually, the message is not suppose to contain it, no? It is a bit annoying because we are losing information without the message here.

nit: You pass two arguments but have only one `%s` in the pattern.

Reply:
I guess it is due to this:
```
    public Throwable(Throwable cause) {
        fillInStackTrace();
        detailMessage = (cause==null ? null : cause.toString());
        this.cause = cause;
    }
``` 
`cause.toString()` contains the stack trace.

Reply:
Thanks for catching that I passed in two arguments!  (That would have prevented this PR from fixing the issue...)

Reply:
Would you have an example of the exception that was used to hit this?

Reply:
Unfortunately, I do not have an example of the exception.  I wish I did.